internal/controllers/cluster/cluster_controller.go

```diff
@@ -216,30 +235,14 @@ func patchCluster(ctx context.Context, patchHelper *patch.Helper, cluster *clust
 	return patchHelper.Patch(ctx, cluster, options...)
 }
 
-// reconcile handles cluster reconciliation.
-func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster) (ctrl.Result, error) {
-	log := ctrl.LoggerFrom(ctx)
-
-	if cluster.Spec.Topology != nil {
-		if cluster.Spec.ControlPlaneRef == nil || cluster.Spec.InfrastructureRef == nil {
-			// TODO: add a condition to surface this scenario
-			log.Info("Waiting for the topology to be generated")
-			return ctrl.Result{}, nil
-		}
-	}
```

Comment:
Is it okay to just drop this code?

EDIT: I think it's not

```
	if s.cluster.Spec.InfrastructureRef == nil {
		// If the infrastructure ref is not set, marking the infrastructure as ready (no-op).
		cluster.Status.InfrastructureReady = true
		conditions.MarkTrue(cluster, clusterv1.InfrastructureReadyCondition)
		return ctrl.Result{}, nil
	}
```

Reply:
Good catch, thank you.